apph-front/src/services/UserService.ts

```diff
@@ -36,6 +36,36 @@ class UserService {
     };
     return Server.request(URL, requestOptions, successFunction, errorFunction);
   }
+
+  static async signUp(
+    email: string,
+    password: string,
+    firstName: string,
+    lastName: string,
+    handleSuccess: () => void,
+    handleError: (errorMessage: string) => void
+  ) {
+    const URL = `${API_URL}/auth/signUp`;
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({
+        email,
+        password: cryptoJS.SHA256(password).toString(),
+        firstName,
+        lastName
+      })
+    };
+    const successFunction = () => {
+      handleSuccess();
+    };
+    const errorFunction = (errorMessage: string) => {
+      handleError(errorMessage);
+    };
+    return Server.request(URL, requestOptions, successFunction, errorFunction);
```

Comment:
Tu peux passer directement tes fonctions
```suggestion
    return Server.request(URL, requestOptions, handleSuccess, handleError);
```